src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java

```diff
@@ -71,8 +77,13 @@ private GtfsRtFactory() {}
 
     public static GtfsRealtime.TripUpdate newTripUpdate(StopEvent event) {
 
+        Pattern routePattern = Pattern.compile(ROUTE_NUMBER_REMOVAL_REGEX);
+        Matcher matcher = routePattern.matcher(event.getRouteData().getRouteName());
+        matcher.find();
+        String parsedRouteName = matcher.group(1);
```

Comment:
This will throw if matcher.find() has returned false. Would it be safer to just use the original routename if find() returns false? Or shall we just throw?

Reply:
I added logic to validate the route name format when validating the Pulsar message.